app/code/Magento/QuoteGraphQl/Model/Resolver/MergeCarts.php

```diff
@@ -31,44 +37,87 @@ class MergeCarts implements ResolverInterface
      */
     private $cartRepository;
 
+    /**
+     * @var CustomerCartResolver
+     */
+    private $customerCartResolver;
+
+    /**
+     * @var QuoteIdToMaskedQuoteIdInterface
+     */
+    private $quoteIdToMaskedQuoteId;
+
     /**
      * @param GetCartForUser $getCartForUser
      * @param CartRepositoryInterface $cartRepository
+     * @param CustomerCartResolver $customerCartResolver
+     * @param QuoteIdToMaskedQuoteIdInterface $quoteIdToMaskedQuoteId
      */
     public function __construct(
         GetCartForUser $getCartForUser,
-        CartRepositoryInterface $cartRepository
+        CartRepositoryInterface $cartRepository,
+        CustomerCartResolver $customerCartResolver,
+        QuoteIdToMaskedQuoteIdInterface $quoteIdToMaskedQuoteId
     ) {
         $this->getCartForUser = $getCartForUser;
         $this->cartRepository = $cartRepository;
+        $this->customerCartResolver = $customerCartResolver;
+        $this->quoteIdToMaskedQuoteId = $quoteIdToMaskedQuoteId;
     }
 
     /**
      * @inheritdoc
      */
-    public function resolve(Field $field, $context, ResolveInfo $info, array $value = null, array $args = null)
-    {
+    public function resolve(
+        Field $field,
+        $context,
+        ResolveInfo $info,
+        array $value = null,
+        array $args = null
+    ) {
         if (empty($args['source_cart_id'])) {
-            throw new GraphQlInputException(__('Required parameter "source_cart_id" is missing'));
-        }
-
-        if (empty($args['destination_cart_id'])) {
-            throw new GraphQlInputException(__('Required parameter "destination_cart_id" is missing'));
+            throw new GraphQlInputException(__(
+                'Required parameter "source_cart_id" is missing'
+            ));
         }
 
         /** @var ContextInterface $context */
         if (false === $context->getExtensionAttributes()->getIsCustomer()) {
-            throw new GraphQlAuthorizationException(__('The current customer isn\'t authorized.'));
+            throw new GraphQlAuthorizationException(__(
+                'The current customer isn\'t authorized.'
+            ));
+        }
+        $currentUserId = $context->getUserId();
+
+        if (empty($args['destination_cart_id'])) {
+            try {
+                $cart = $this->customerCartResolver->resolve($currentUserId);
+            } catch (CouldNotSaveException $exception) {
+                throw new GraphQlNoSuchEntityException(
+                    __('Could not create empty cart for customer'),
+                    $exception
+                );
+            }
+            $customerMaskedCartId = $this->quoteIdToMaskedQuoteId->execute(
+                (int) $cart->getId()
+            );
         }
 
         $guestMaskedCartId = $args['source_cart_id'];
-        $customerMaskedCartId = $args['destination_cart_id'];
+        $customerMaskedCartId = $customerMaskedCartId ?? $args['destination_cart_id'];
```

Comment:
There's magic in 7.4 that allows us to turn it into 

```suggestion
        $customerMaskedCartId ??= $args['destination_cart_id'];
```

But let's leave it as it is to preserve compatibility with 7.3